### ISSUE TITLE
Is the process of latent states finding differentiable?

### ISSUE THREAD

**Issue by wadx2019** [locuslab/deq-ddim#2]
According to the paper and code, deq-ddim-inversion seems unable to differentiate from x^T to the known image x^0. Thus, I am curious whether there exists a way to calculate the Jacobian matrix from x^T to x^0.

**Comment by ashwinipokle** [comment]
Hello, Thank you for interest in our work. Given a known image x^0, DEQ-DDIM enables you to optimize a random x^T so that it can generate that x^0. This process involves calculating Jacobian of x^T w.r.t x^0 (internally through PyTorch autograd) for gradient descent updates. Please check relevant pieces of code for diffusion inversion [here](https://github.com/locuslab/deq-ddim/blob/main/runners/diffusion_inversion.py) and [here](https://github.com/locuslab/deq-ddim/blob/main/functions/latent_space_opt_anderson.py) . 

Specifically, as you can see in [Line 345](https://github.com/locuslab/deq-ddim/blob/main/runners/diffusion_inversion.py#L345), calculating loss and doing backward(), involves backprop through entire DEQ-DDIM, so you should be able to extract required gradients. 

Let me know if you have any further questions.
